addon/-private/system/many-array.js

```diff
@@ -87,15 +87,39 @@ export default Ember.Object.extend(Ember.MutableArray, Ember.Evented, {
     );
     toSet = toSet.concat(newRecords);
     var oldLength = this.length;
-    this.arrayContentWillChange(0, this.length, toSet.length);
-    // It’s possible the parent side of the relationship may have been unloaded by this point
-    if (_objectIsAlive(this)) {
-      this.set('length', toSet.length);
+    var newLength = toSet.length;
+
+    var shortestLength = Math.min(oldLength, newLength);
+
+    var firstChangeIndex = -1; // -1 signifies no changes
+    // find the first change
+    var currentArray = this.currentState;
+    for (var i=0; i<shortestLength; i++) {
```

Comment:
we should likely skip this if within `init`, but that isn't needed i think until https://github.com/emberjs/data/pull/4600 is merged, so either this one or that one need to take this into account.